app/src/composables/useSiwe.ts

```diff
@@ -1,175 +1,149 @@
-import { SLSiweMessageCreator } from '@/adapters/siweMessageCreatorAdapter'
 import router from '@/router'
 import { ref, watch } from 'vue'
 import { useUserDataStore, useToastStore } from '@/stores'
 import type { User } from '@/types'
-import { log, parseError } from '@/utils'
+import { log } from '@/utils'
 import { useCustomFetch } from './useCustomFetch'
 import { useStorage } from '@vueuse/core'
-import { useAccount, useSignMessage, useConnect, useSwitchChain } from '@wagmi/vue'
-import { NETWORK } from '@/constant'
-
-function createSiweMessageCreator(address: string, statement: string, nonce: string | undefined) {
-  return new SLSiweMessageCreator({
-    address,
-    statement,
-    nonce,
-    version: '1',
-    chainId: 1
-  })
-}
+import { useAccount, useSignMessage, useChainId } from '@wagmi/vue'
+import { SiweMessage } from 'siwe'
+import { useWalletChecks } from '@/composables'
 
 export function useSiwe() {
-  const { addErrorToast } = useToastStore()
-  const isProcessing = ref(false)
+  //#region Refs
   const authData = ref({ signature: '', message: '' })
   const apiEndpoint = ref<string>('')
-  const { connectors, connect, error: connectError } = useConnect()
-  const { address, isConnected } = useAccount()
-  const { switchChain } = useSwitchChain()
-
-  watch(connectError, (newVal) => {
-    if (newVal) {
-      addErrorToast(parseError(newVal))
-      log.error('connectError.value', newVal)
-      isProcessing.value = false
-    }
-  })
-
-  watch(address, async (newVal) => {
-    if (newVal) {
-      console.log(`connected address: `, newVal)
-      await executeSiwe()
-    }
-  })
+  //#endregion
 
+  //#region Composables
+  const { addErrorToast } = useToastStore()
+  const userDataStore = useUserDataStore()
+  const { address } = useAccount()
+  const chainId = useChainId()
```

Comment:
You should use signMessageAsync instead of signMessage. This allow you to wait for the signature.
const { data: signature, error: signMessageError, signMessageAsync} = useSignMessage()

Reply:
And by using this you won't need to use a watcher to watch the signature, you can simply process to the next step.